libs/dh/admin/feature-user-management/src/lib/feature-create-userrole/tabs/dh-create-userrole-tabs.component.ts

```diff
@@ -0,0 +1,179 @@
+  providers: [
+    provideComponentStore(DhAdminUserRolesManagementDataAccessApiStore),
+  ],
+  styles: [
+    `
+      :host {
+        display: block;
+      }
+    `,
+  ],
+  imports: [
+    TranslocoModule,
+    CommonModule,
+    PushModule,
+    LetModule,
+    WattTabsModule,
+    WattValidationMessageModule,
+    DhCreateUserroleMasterdataTabComponent,
+    DhCreateUserrolePermissionsTabComponent,
+    WattCardModule,
+    WattSpinnerModule,
+    WattButtonModule,
+    ReactiveFormsModule,
+    FormsModule,
+  ],
+})
+export class DhCreateUserroleTabsComponent implements OnInit, OnDestroy {
+  private readonly store = inject(DhAdminUserRolesManagementDataAccessApiStore);
+  isLoading$ = this.store.isLoading$;
+  validation$ = this.store.validation$;
+  createRoleForm = this.fb.group<CreateRoleForm>({});
+  userRole?: CreateUserRoleDto;
+
+  private destroy$ = new Subject<void>();
+
+  constructor(
+    private router: Router,
+    private fb: FormBuilder,
+    private toastService: WattToastService,
+    private translocoService: TranslocoService
+  ) {}
+
+  ngOnInit(): void {
+    this.userRole = {
+      name: '',
+      description: '',
+      eicFunction: 'Agent',
+      status: 'Active',
+      permissions: [],
+    };
+
+    this.store.createRequestHasError$
+      .pipe(takeUntil(this.destroy$))
+      .subscribe((hasError) => {
+        if (hasError) {
+          this.toastService.open({
+            message: this.translocoService.translate(
+              'charges.createPrices.createPriceError'
+            ),
+            type: 'danger',
+          });
+
+          this.enableControls();
+        }
+      });
+  }
+
+  ngOnDestroy(): void {
+    this.destroy$.next();
+    this.destroy$.unsubscribe();
+  }
+
+  onEicFunctionSelected = (eic: EicFunction) => {
+    console.log(eic);
+  };
+
+  onSubmit() {
+    if (!this.userRole) throw new Error('Missing user role');
+    this.disableControls();
+    this.store.createUserRole({
+      createRole: this.userRole,
+      onSaveCompletedFn: this.backToOverview,
+    });
+
+    this.toastService.open({
+      message: this.translocoService.translate(
+        'admin.userManagement.createrole.createRoleRequestText'
+      ),
+      type: 'loading',
+    });
+  }
+
+  disableControls() {
+    this.createRoleForm.controls.masterData?.controls.description.disable();
+    this.createRoleForm.controls.masterData?.controls.eicFunction.disable();
+    this.createRoleForm.controls.masterData?.controls.name.disable();
+    this.createRoleForm.controls.masterData?.controls.roleStatus.disable();
+  }
+
+  enableControls() {
+    this.createRoleForm.controls.masterData?.controls.description.enable();
+    this.createRoleForm.controls.masterData?.controls.eicFunction.enable();
+    this.createRoleForm.controls.masterData?.controls.name.enable();
+    this.createRoleForm.controls.masterData?.controls.roleStatus.enable();
+  }
+
+  addChildForm<K extends keyof CreateRoleForm>(
```

Comment:
Question: Where is this used, I cannot find reference for it anywhere

Reply:
It's used when the sub components are emitting that they are ready, so they are added to the formgroup